torch/_subclasses/fake_tensor.py

```diff
@@ -1509,6 +1531,15 @@ def _prep_args_for_hash(
                 raise _BypassDispatchCache("symbolic shape")
             elif isinstance(arg, (list, tuple, dict)):
                 self._prep_args_for_hash(result, arg, state)
+            elif isinstance(arg, types.FunctionType):
+                raise _BypassDispatchCache("function argument")
+            elif isinstance(arg, torch.fx.GraphModule):
+                # This is used for invoke_subgraph where id(graph_module) allows
+                # us to cache fake outputs
+                result.append(type(arg))
+                result.append(id(arg))
+            elif isinstance(arg, FunctionalizeCtxWrapper):
+                result.append(hash(arg))
```

Comment:
umm, does id(arg) assume that the GraphModule stays alive forever? (What if the GraphModule gets deallocated and another one gets allocated in its stead?)

We might need some cache invalidation mechanism via weakref.finalize

Reply:
Doing it here - https://github.com/pytorch/pytorch/pull/149667